fix(sidebar): use open state to remove always working focus trap on mobile

ShGKme opened the pull request:
### ☑️ Resolves

* Fix: https://github.com/nextcloud/spreed/issues/11454
* Bump `@nextcloud/vue` from 8.11.3 to 8.13.0
* Using new `open` state of `NcAppSidebar` with built-in toggle button
* Adjusted animations

<!--
░░░░░░░░░░░░░░░░░░░░
░█████░░█████░█████░
░░███░░░░███░░░███░░
░░███░░░░███░░░███░░
░░███░░░░███░░░███░░
░░░████████░░░█████░
░░░░░░░░░░░░░░░░░░░░

Feel free to remove this section when your PR is only touching backend/API code
-->

## 🖌️ UI Checklist

### 🖼️ Screenshots / Screencasts

🏚️ Before | 🏡 After
-- | --
![image](https://github.com/nextcloud/spreed/assets/93392545/3c1e3105-97c9-4569-a2ee-d6c68cd60a5f) | ![image](https://github.com/nextcloud/spreed/assets/93392545/7115f28b-da6e-4566-8460-ab4ba98cc0b9)


<!-- ☀️ Light theme | 🌑 Dark Theme -->

### 🚧 Tasks

- [ ] Style changes

### 🏁 Checklist

- [ ] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [ ] 🖥️ Tested with Desktop client or should not be risky for it 
- [ ] 🖌️ Design was reviewed, approved or inspired by the design team
- [ ] ⛑️ Tests are included or not possible
- [ ] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

Antreesy reviewed (approved):
Works as expected. Though there are still too many styles changes came from library (follow-up?)

ShGKme commented:
> Though there are still too many styles changes came from library (follow-up?)

What do you mean? Are they from the library or new server styles?

ShGKme commented:
`@nextcloud/vue@8` doesn't suppose to have any breaking change at least for supported 28 and 29.

Styles on 30 are broken in many places also from server styles, it's in progress

Antreesy commented:
Yeah, server styles, I have compared at `main` only. Tested with desktop + server 29, so far looks the same

ShGKme commented:
- Rebased onto `main`
- Minor fixes in fixup:
  - Correctly use conditional slot
  - Pass all the previous attrs to the button like title and aria-label.

Antreesy commented on src/components/RightSidebar/RightSidebar.vue:
```diff
@@ -156,7 +175,7 @@ export default {
 		},
 		opened() {
-			return !!this.token && !this.isInLobby && this.show
+			return !this.isInLobby && this.show
```
Makes sense because of `v-if="token"`

Antreesy commented on src/components/RightSidebar/RightSidebar.vue:
```diff
@@ -4,14 +4,27 @@
+		<template v-if="isInCall" #toggle-icon>
+			<MessageText :size="20" />
+			<NcCounterBubble v-if="unreadMessagesCounter > 0"
+				class="chat-button__unread-messages-counter"
```
![image](https://github.com/nextcloud/spreed/assets/93392545/5b542008-7ca6-4c77-b566-ff1214f309f2)

we don't need to bury class under :deep selector, I think

ShGKme replied:
For `chat-button__unread-messages-counter` deep is not needed, but for the `chat-button` itself it is needed.

Let me check with 29 styles...

ShGKme replied:
Working on a regression

ShGKme replied:
Thanks for noticing the problem

ShGKme commented:
- Rebased onto `main` and resolved conflicts
- Fixed styles. Required, because with new Teleport fix in `NcAppSidebar`, the toggle is no longer inside the sidebar

![image](https://github.com/nextcloud/spreed/assets/25978914/c06c115a-1b03-44b5-a5f2-10723cac698d)

ShGKme commented:
Note, screenshot is made on stable29 styles

Antreesy reviewed (approved):
Tested against stable29

on main ubread marker is still messy, but because of reducing the clickable area 44 -> 34

DorraJaouad commented:
On main :
message (shouldn't be lower ? It doesn't look much readable)
![image](https://github.com/nextcloud/spreed/assets/84044328/dd418940-0bed-4cd6-b64d-042086eec192)
mention (at all) 
![image](https://github.com/nextcloud/spreed/assets/84044328/2a2e53ef-d249-486a-9024-55f85ce5ff4e)
direct mention 
![image](https://github.com/nextcloud/spreed/assets/84044328/9ac30278-9c05-49d6-af09-00d8dd63df8e)

Color should be applied only on message notif
```
.chat-button__unread-messages-counter {
	color: var(--color-primary-element);
}
```
Besides, styles should be aligned with conversation counter styles ?

ShGKme commented:
> On main

@DorraJaouad Is it much different from what was before? Styles on main, including `@nextcloud/vue` are not adjusted to new theming.

Antreesy commented on src/components/RightSidebar/RightSidebar.vue:
```diff
@@ -391,4 +484,23 @@ export default {
+	bottom: 2px;
+	right: 2px;
+	pointer-events: none;
+	color: var(--color-primary-element);
```
On stable29 | With a patch
-- | --
![image](https://github.com/nextcloud/spreed/assets/93392545/6de4d757-4e26-47c6-854b-21794cd09d99) | ![image](https://github.com/nextcloud/spreed/assets/93392545/79338995-3df3-4492-8f44-bdc3be6cc6ec)

```suggestion

	&.counter-bubble__counter--highlighted {
		color: var(--color-primary-text);
	}
```

Antreesy replied:
This hasn't changed since Feb.2023 https://github.com/nextcloud/spreed/commit/d3a062e25ea770dc62df4957beb58f487fe3fb98, and not related to this PR changes, so could be fixed in follow-up

Antreesy commented:
Rebased on `main`, squashed fixups and provide a fix for 'highlighted' type:
![image](https://github.com/nextcloud/spreed/assets/93392545/3122e581-413d-46d2-b528-47c6b0715ce4)

DorraJaouad reviewed (approved):
Tested

ShGKme commented:
/backport to stable29
